File: 0977-squares-of-a-sorted-array/0977-squares-of-a-sorted-array.py
class Solution:
    def sortedSquares(self, nums: List[int]) -> List[int]:
        left = 0
        right = len(nums) - 1
        pos = len(nums) - 1
        new_nums = [0] * len(nums)

        while left <= right:
            if abs(nums[left]) > abs(nums[right]):
                new_nums[pos] = nums[left] ** 2
                left += 1
            else:
                new_nums[pos] = nums[right] ** 2
                right -= 1
            pos -= 1

        return new_nums

File: 0977-squares-of-a-sorted-array/0977-squares-of-a-sorted-array.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
// var sortedSquares = function (nums) {
//     for (let i = 0; i < nums.length; i++) {
//         nums[i] = nums[i] * nums[i];
//     }
//     nums.sort((a, b) => a - b);
//     return nums;
// };

var sortedSquares = function (nums) {
    let left = 0;
    let right = nums.length - 1;

    let result = [];
    while (left <= right) {
        if (Math.abs(nums[left]) > Math.abs(nums[right])) {
            result.push(nums[left] * nums[left]);
            left++;
        } else {
            result.push(nums[right] * nums[right]);
            right--;
        }
    }
    // console.log(result);

    return result.reverse();
}